webapp/src/components/App/Singlestudy/explore/TableModeList/utils.ts

```diff
@@ -37,7 +37,22 @@ export function createTableTemplate<T extends TableModeType>(
   return { id: uuidv4(), name, type, columns };
 }
 
-export function getTableColumnsForType(type: TableModeType): readonly string[] {
-  // Arrays have a numeric index signature because of `as const`
-  return TABLE_MODE_COLUMNS_BY_TYPE[type];
+/**
+ * Returns the columns for the specified table type and study version.
+ *
+ * @param tableType Table type ("areas", "links", ...).
+ * @param studyVersion Study version number.
+ */
+export function getTableColumnsForType(
+  tableType: TableModeType,
+  studyVersion?: number,
+): string[] {
+  const columnsByVersion = TABLE_MODE_COLUMNS_MAPPING[tableType];
+  let result: string[] = [];
+  for (const sinceVersion in columnsByVersion) {
+    if (studyVersion == undefined || studyVersion >= Number(sinceVersion)) {
```

Comment:
In TypeScript/JavaScript, the logical OR operator `||` is evaluated lazily: it only evaluates the second arg if the first one evaluates to false. So, if `studyVersion` is undefined, the second part is not evaluated here.
In this function, if the `studyVersion` parameter is not provided, it defaults to `undefined`, and we want to retrieve all columns regardless of the version.

Reply:
I know, you can also note if both are falsy, the last value is returned, javascript can be tricky sometimes 😄 
my point was not about the operator used but rather the intent of your `if` condition, let me explain more:

when designing functions, especially those involved in data retrieval / manipulation like `getTableColumnsForType`, it’s crucial to clearly differentiate between distinct use cases. 

```javascript
if (studyVersion === undefined || studyVersion >= Number(version))
```
here the combined check in the conditional statement mixes two fundamentally different scenarios into one condition, which can lead to several potential issues:
- it reduce the intent and clarity, making it harder to understand and debug
- if someone touch it in the future mixed use cases in a condition can introduce hard to find bugs
etc...

to avoid this you can check for undefined before the loop, and so handle this case in isolation, example:
```javascript
if(!studyVersion){ // true for undefined, null, 0, "", NaN, false
// return all columns
}
// else we handle the versioning of columns
```
also be careful with the `==` operator generally it's better to use the strict comparison operator to avoid coercion and so unexpected types conversions. (you can use the `!` operator in most cases to check the absence of a value)

or better handle this directly at the data source (like in the provided [example](https://github.com/AntaresSimulatorTeam/AntaREST/pull/1643/files/c1b7a77b40376fa39e61832761f2217e23bbdafb#diff-280fd41713233f4d7e7ef9bd69fb798391b6d30ef738e98372680e201c6172ce) below)